### PR TITLE
Add Constraints For Issue #798

### PR DESCRIPTION
**Pull request by Gabeblis**
# Committer Notes
## Purpose
The purpose of this PR is to add the constraints for issue #798 and demonstrate the manual approach for these constraints. There are 2 approaches: 1) Make an individual constraint for each policy/procedure check. 2) Make 2 tests (`has-policy` and `has-procedure`) and use a map to store the messages to make the constraints more dynamic and minimize constraints and files added. 

## Changes
- Added all of the policy/process-procedure constraints from the [table](https://github.com/GSA/fedramp-automation/issues/809#issuecomment-2522112893) in the comments on Issue #809. 
- Added negative content for the constraints and edited `fedramp-ssp-example.oscal.xml` to align with the constraints.
- Added pass/fail yaml files for each constraint.

### All Submissions:

- [x] Have you selected the correct base branch per [Contributing](https://github.com/GSA/fedramp-automation/blob/master/CONTRIBUTING.md) guidance?
- [x] Have you set "[Allow edits and access to secrets by maintainers](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork)"?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/GSA/fedramp-automation/pulls) for the same update/change?
- [ ] Have you squashed any non-relevant commits and commit messages? \[[instructions](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History)\]
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] If applicable, have all [FedRAMP Documents Related to OSCAL Adoption](https://github.com/GSA/fedramp-automation) affected by the changes in this issue have been updated.?
- [x] If applicable, does this PR reference the issue it addresses and explain how it addresses the issue?

By submitting a pull request, you are agreeing to provide this contribution under the [CC0 1.0 Universal public domain](https://creativecommons.org/publicdomain/zero/1.0/) dedication.

### CODE REVIEW

**Review by aj-stein-gsa — COMMENTED** [review]
I am still working through the index-based approach, but this is very nice. I am curious to hear what the rest of the team thinks.

**Comment by wandmagic** [comment]
I would prefer the index based approach, however,  this does work. 
My main concern is we are inflating the time it takes to run make test considerably for little tangible benefit. 
I would be ok with this approach if we take all INVALID content into a single file so we can leverage the cache and reduce impact on unit test performance.

**Review by wandmagic — CHANGES_REQUESTED** [review]
lets use a single INVALID file for these constraints

**Comment by wandmagic** [comment]
looks like the runtime is a little lower, but I'm starting to think i need to improve performance again. i'll address that in another PR.

**Review by wandmagic — APPROVED** [review]

**Comment by aj-stein-gsa** [comment]
> looks like the runtime is a little lower, but I'm starting to think i need to improve performance again. i'll address that in another PR.

This is what I was worried about, can you elaborate, @wandmagic? The slowness I noticed last week seemed to be vastly improved by use of a resolved catalog (so a no-op on `resolve-profile()` and loading a local catalog/profile from a relative bath, forgoing `https://...` lookups. Is it more signficant than that? And in either case, how can we measure that better?

**Comment by Rene2mt** [comment]
Nice work @Gabeblis.  I like "Approach # 2"....they all follow the exact same pattern.  Approach 2 is leaner, achieving the same desired behavior with less code.  It will be more maintainable long-term.

**Comment by aj-stein-gsa** [comment]
> I am still working through the index-based approach, but this is very nice. I am curious to hear what the rest of the team thinks.

Given your speed on this and yesterday's bug discoveries (and the limits of cleverness around indices), I think we are moving forward with one of your approaches, @Gabeblis, congratulations. 😅  🎉 I shall start reviewing.

**Review by aj-stein-gsa — DISMISSED** [review]
I think this is a good start, let's move forward with Option 2 then?

**Comment by Gabeblis** [comment]
> I think this is a good start, let's move forward with Option 2 then?

Sounds like that's the direction we're going. I'll update the PR now.

**Review by aj-stein-gsa — DISMISSED** [review]
I know it needs a rebase, but this looks good for now.

File: src/validations/constraints/content/ssp-has-policy-and-procedure-INVALID.xml
Comment (aj-stein-gsa) [inline src/validations/constraints/content/ssp-has-policy-and-procedure-INVALID.xml:1]: Can we slim this down with less examples to not cause a conflict with the recently merged STYLE PR?

**Review by aj-stein-gsa — COMMENTED** [review]
I know one issue that will come up we can discuss, but otherwise, this one looks good. I just know it will fail the build once we merge.

File: src/validations/constraints/content/ssp-has-policy-and-procedure-INVALID.xml
Comment (Gabeblis) [reply src/validations/constraints/content/ssp-has-policy-and-procedure-INVALID.xml:1]: Done 🫡

**Review by aj-stein-gsa — APPROVED** [review]

**Review by wandmagic — APPROVED** [review]

**Pull request merged**